### PR TITLE
Creates TestReportDir if not exists

### PR DESCRIPTION
**Pull request by zionyx**
* Renamed `New-RandomFolder` > `New-RandomFolderForTestUse`.
* Supports relative path as TestReportDir.
* Creates TestReportDir if it does not exist.

### CODE REVIEW

**Comment by casz** [comment]
Please resolve conflicts 😓

File: Test-Source/Utilities.Tests.ps1 [hunk 1]
```diff
@@ -6,15 +6,15 @@ Describe 'Validate various support functions for testing' {
     Context 'Validating support functions for - When git is installed' {
 
         It 'can create random folder in TEMP folder' {
-            $pathInTemp = New-RandomFolder
+            $pathInTemp = New-RandomFolderForTestUse
```
Comment (rasmusjelsgaard) [inline Test-Source/Utilities.Tests.ps1]: Please see my previous comment regarding this one.

File: Source/Private/Utilities.ps1 [hunk 2]
```diff
@@ -1,4 +1,4 @@
-function New-RandomFolder {
+function New-RandomFolderForTestUse {
```
[the remark this answers is not in the document]
Comment (zionyx) [reply Source/Private/Utilities.ps1]: Functionality moved to `New-RandomFolder.ps1`, tests moved to its associated test file.

**Comment by zionyx** [comment]
Rebased to master.

**Pull request merged**